diff --git a/core/modules/sdc/src/Twig/ComponentNodeVisitor.php b/core/modules/sdc/src/Twig/ComponentNodeVisitor.php
index e7f2a928a5c56b1ca9d21bf545425bda99a0a1e8..b6ad7a2ff9a503e129a0fb03503ed00a43d78dcc 100644
--- a/core/modules/sdc/src/Twig/ComponentNodeVisitor.php
+++ b/core/modules/sdc/src/Twig/ComponentNodeVisitor.php
@@ -71,15 +71,25 @@ public function leaveNode(Node $node, Environment $env): ?Node {
       new Node([new ConstantExpression($component_id, $line)]),
       $line
     ), $line);
-    foreach ($print_nodes as $index => $print_node) {
-      $node->getNode('display_start')->setNode((string) $index, $print_node);
-    }
+
+    // Append the print nodes to the display_start node.
+    $node->setNode(
+      'display_start',
+      new Node(array_merge([
+        $node->getNode('display_start')],
+        $print_nodes)
+      )
+    );
+
     if ($env->isDebug()) {
-      $node->getNode('display_end')
-        ->setNode(
-          '0',
-          new PrintNode(new ConstantExpression(sprintf('<!-- %s Component end: %s -->', $emoji, $component_id), $line), $line)
-        );
+      // Append the closing comment to the display_end node.
+      $node->setNode(
+        'display_end',
+        new Node([
+          new PrintNode(new ConstantExpression(sprintf('<!-- %s Component end: %s -->', $emoji, $component_id), $line), $line),
+          $node->getNode('display_end')]
+        )
+      );
     }
     // Slots can be validated at compile time, we don't need to add nodes to
     // execute functions during display with the actual data.
